src/content/manager/package.manager.ts

```diff
@@ -91,26 +100,43 @@ export class PackageManager extends BaseManager {
     }
 
     private buildPushUrl(): string {
-        let pushUrl = `${PackageManager.BASE_URL}/${PackageManager.IMPORT_ENDPOINT_PATH}`;
-        if (this.newKey) {
-            pushUrl = `${pushUrl}?newKey=${this.newKey}`;
-        }
-        if (this.overwrite) {
-            if (this.newKey) {
-                logger.error(
-                    "You cannot overwrite a package and set a new key at the same time. Please use only one of the options."
-                );
-                process.exit();
-            } else {
-                pushUrl = `${pushUrl}?overwrite=${this.overwrite}`;
-            }
-        }
-        return pushUrl;
+        this.validateOptions();
+        const pushUrl = `${PackageManager.BASE_URL}/${PackageManager.IMPORT_ENDPOINT_PATH}`;
+        return this.getPushUrlWithParams(pushUrl);
     }
 
     private listPackages(nodes: SaveContentNode[]): void {
         nodes.forEach(node => {
             logger.info(`${node.name} - Key: "${node.key}"`);
         });
     }
+
+    private validateOptions(): void {
+        let hasErrors: boolean;
+        if (!this.spaceKey) {
+            logger.error("You cannot push a package without specifying the space. Please try again.");
+            hasErrors = true;
+        }
+        if (this.newKey && this.overwrite) {
+            logger.error(
+                "You cannot overwrite a package and set a new key at the same time. Please use only one of the options."
+            );
+            hasErrors = true;
+        }
+
+        if (hasErrors) {
+            process.exit();
+        }
+    }
+
+    private getPushUrlWithParams(pushUrl: string): string {
+        let pushUrlWithParams = `${pushUrl}?spaceId=${this.spaceKey}&`;
+        if (this.newKey) {
+            pushUrlWithParams = `${pushUrlWithParams}newKey=${this.newKey}&`;
```

Comment:
nit: Let's return on both if statements because we don't want to have both the newKey and overwrite request params. We also validate this case on the above method.